### PR TITLE
Bump Nessie from 0.43.0 to 0.44.0

### PR DESCRIPTION
**Pull request by ajantha-bhat**
Test plan - CI

```
== RELEASE NOTES ==

Iceberg Changes
* Bump Nessie from 0.43.0 to 0.44.0
```

Major Nessie improvements/changes:
Nessie GC: Command line tool is added.

More details:
https://github.com/projectnessie/nessie/blob/main/site/docs/try/releases.md#0440-release-october-18-2022

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
cc: @snazy, @nastra

File: presto-iceberg/pom.xml
Comment (nastra) [inline presto-iceberg/pom.xml:16]: `NessieContainer` should probably also be updated

Comment (ajantha-bhat) [reply presto-iceberg/pom.xml:16]: It was not bumped for the previous Nessie release. 

So, I assumed, it has to be bumped with the Iceberg update. 

I have updated it now anyways.  I think it is still compatible. waiting for CI result.

**Comment by ajantha-bhat** [comment]
seems to be environment is flaky
> TestHiveQueriesUsingJSON.testSubqueries:16->TestHiveQueries.testSubqueries:634->AbstractTestQueryFramework.assertQueryFails:251 Expected exception message 'Failed to fetched data from 127.0.0.1:1235 /v1/task/20221121_141127_00174_dydwx.2.0.0/results/0/0 - Exhausted retries: Received HTTP 500 std::exception'

**Review by nastra — APPROVED** [review]
LGTM, @beinan could you review this one please? We'd need to bump this before getting Iceberg 1.1.0 in (https://github.com/prestodb/presto/pull/18709)

File: presto-iceberg/pom.xml
Comment (ajantha-bhat) [reply presto-iceberg/pom.xml:16]: The Latest Nessie version is compatible with both Iceberg 1.0.0.
https://github.com/projectnessie/nessie#compatibility

So, it is ok to bump the test containers.

**Review by ajaygeorge — APPROVED** [review]
LGTM.
Release notes section needs to be formatted properly. 
See https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines

**Comment by ajantha-bhat** [comment]
@ajaygeorge: Thanks for the review. I have formatted the release notes section now.

**Comment by ajaygeorge** [comment]
@ajantha-bhat Looks good !

**Comment by ajaygeorge** [comment]
Adding committer @tdcmeehan  to take a look.

**Review by ChunxuTang — APPROVED** [review]
LGTM.
@ajantha-bhat Thanks for your first PR to Presto!

**Comment by ajantha-bhat** [comment]
Can we please merge this PR?
As https://github.com/prestodb/presto/pull/18709 depends on this PR

**Pull request merged**
